src/cmd/manage_nsfs.js

```diff
@@ -355,13 +356,15 @@ async function update_bucket(data) {
 
 async function delete_bucket(data) {
     await validate_bucket_args(data, ACTIONS.DELETE);
-
     const fs_context = native_fs_utils.get_process_fs_context(config_root_backend);
     const bucket_config_path = get_config_file_path(buckets_dir_path, data.name);
     try {
+        const bucket_temp_dir_path = path.join(data.path, config.NSFS_TEMP_DIR_NAME + "_" + data._id);
+        await native_fs_utils.delete_bucket_temp_dirs(fs_context, bucket_temp_dir_path);
         await native_fs_utils.delete_config_file(fs_context, buckets_dir_path, bucket_config_path);
     } catch (err) {
         if (err.code === 'ENOENT') throw_cli_error(ManageCLIError.NoSuchBucket, data.name);
+        if (err.code === 'BUCKET_CLEANUP_FAILED') throw_cli_error(ManageCLIError.BucketCleanUpFailed, data.name);
```

Comment:
2 comments on this line - 
1. no need to do this conversion, rpc_errors will be converted to ManageCLIError errors on the main function, just add data.name to the RPCError.
2. looks like there is a bug here - we need to add `throw err;` at the end of the catch scope

Reply:
@romayalon how can we set data.name in error?

Reply:
you can add it to the RPC error, I added a comment there